internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -844,47 +827,69 @@ func (r *AgentServiceConfigReconciler) newAssistedServiceDeployment(log logrus.F
 		},
 	}
 
-	if instance.Spec.MirrorRegistryRef != nil {
-		mirrorVolumeMounts := []corev1.VolumeMount{
-			{Name: mirrorRegistryCertVolume, MountPath: common.MirrorRegistriesCertificatePath, SubPath: common.MirrorRegistriesCertificateFile},
-			{Name: mirrorRegistryConfVolume, MountPath: common.MirrorRegistriesConfigDir},
-		}
-		mirrorVolumes := []corev1.Volume{
-			{
-				Name: mirrorRegistryCertVolume,
-				VolumeSource: corev1.VolumeSource{
-					ConfigMap: &corev1.ConfigMapVolumeSource{
-						LocalObjectReference: *instance.Spec.MirrorRegistryRef,
-						DefaultMode:          swag.Int32(420),
-						Optional:             swag.Bool(true),
-						Items: []corev1.KeyToPath{
-							{
-								Key:  mirrorRegistryRefCertKey,
-								Path: common.MirrorRegistriesCertificateFile,
+	if mirrorRegConfigMap != nil {
```

Comment:
Unless I'm missing something, I think will always be true since there's `mirrorCM := &corev1.ConfigMap{}` in L297, which is then passed to the call to this function. So, it will never be nil.

We have to options: Receive `mirrorRegConfigMap` by value and keep things as they are or change L297 to something else :rofl: 

https://play.golang.org/p/1zGukCXpgYX

Reply:
Let me know what you think of how I solved this.